### ISSUE TITLE
registerSingleton dispose not called on unregister

### ISSUE THREAD

**Issue by bogdanmarinescu** [fluttercommunity/get_it#172]
```
class MyBloc {
  void dispose() {
    print('dispose MyBloc');
  }
}

void main() {
  GetIt.I.registerSingleton<MyBloc>(MyBloc(), dispose: (b) {
    b.dispose();
  });

  GetIt.I.unregister<MyBloc>();
}
```
On unregister the registerSingleton `dispose` callback should be called.
This bug #138 was closed in  5.0.2, but we still have it in 6.0.0.

**Comment by escamoteur** [comment]
Yep, that somehow got lost, sorry

**Comment by escamoteur** [comment]
Implemented in V6.1.0
